src/openrct2/world/Park.cpp

```diff
@@ -516,7 +517,13 @@ money32 Park::CalculateRideValue(const Ride* ride) const
 
 money32 Park::CalculateCompanyValue() const
 {
-    return finance_get_current_cash() + gParkValue - gBankLoan;
+    money32 result = finance_get_current_cash();
+
+    // Clamp addition and substraction to prevent overflow
+    result = add_clamp_money32(result, -gBankLoan);
+    result = add_clamp_money32(result, gParkValue);
```

Comment:
This might alter the outcome and I would like to know why it overflows in the first place.

I'm going to assume `finance_get_current_cash()` can only return positive values or zero.

The problem is you can have maxed bank loan, `gBankLoan` much larger than `finance_get_current_cash()`, but it will get clamped and then `gParkValue > 0`, which would skew the calculated company value. Previously such case _could_ wrap twice (both ways) and end up being correct.

This is just my speculation, I don't know if this can actually happen.

For parks that wrap the value, what's the most probable reason? Wouldn't it make more sense to calculate `pvbl = gParkValue - gBankLoad` first and then add `result = current_cash + pvbl`?

Reply:
I believe `finance_get_current_cash()` returns negative values as well:
![image](https://user-images.githubusercontent.com/40113382/77483733-2adc0c00-6e29-11ea-92ce-957e75b533f0.png)

I am not sure if I follow the part about `gParkValue`. The `CalculateParkValue()` method, which sets `gParkValue`, only takes the number of guests and the ride values into account. So, I don't see how that would skew the calculated company value.

If `gBankLoan` is much larger than `finance_get_current_cash()`, then the company value will be clamped to `INT32_MIN` instead of overflowing.

Reply:
Also, the reasoning behind why it overflows in the first place is as follows. Given that `result = finance_get_current_cash() + pvbl` with `pvbl = gParkValue - gBankLoan` . So:

1. Use cheat to max cash: `finance_get_current_cash() = INT32_MAX`
2. A situation arises where `pvbl = gParkValue - gBankLoan > 0`. This can happen when the park value is updated when a ride is opened (which explains why it overflows as seen in #11005).
3. Then, `result = INT32_MAX + pvbl` and since `pvbl > 0`, it must hold that `result > INT32_MAX`. Thus, overflow happens.